mlflow/cli.py

```diff
@@ -510,8 +521,35 @@ def gc(backend_store_uri, run_ids):
         raise MlflowException(
             "This cli can only be used with a backend that allows hard-deleting runs"
         )
+
+    current_time = int(time.time() * 1000)
+
+    if older_than != "0":
+        regex = re.compile(
+            r"^((?P<days>[\.\d]+?)d)?((?P<hours>[\.\d]+?)h)?((?P<minutes>[\.\d]+?)m)?((?P<seconds>[\.\d]+?)s)?$"
+        )
+        parts = regex.match(older_than)
+        if parts is None:
+            raise MlflowException(
+                "Could not parse any time information from '{}'.  Examples of valid strings: '8h', '2d8h5m20s', '2m4s'".format(
+                    older_than
+                )
+            )
+        time_params = {name: float(param) for name, param in parts.groupdict().items() if param}
+        time_delta = int(timedelta(**time_params).total_seconds() * 1000)
+
     if not run_ids:
-        run_ids = backend_store._get_deleted_runs()
+        if older_than != "0":
+            # filter run_ids based on older_than parameter
+            _run_ids = backend_store._get_deleted_runs()
```

Comment:
Can we introduce an optional `older_than` parameter to `_get_deleted_runs()`? This will improve efficiency for the SQLAlchemyStore, and it will also improve code readability here.